torch/csrc/jit/tensorexpr/ir_cloner.cpp

```diff
@@ -0,0 +1,343 @@
+#include <torch/csrc/jit/tensorexpr/ir_cloner.h>
+
+#include <torch/csrc/jit/tensorexpr/eval.h>
+#include <torch/csrc/jit/tensorexpr/ir.h>
+#include <torch/csrc/jit/tensorexpr/ir_simplifier.h>
+#include <torch/csrc/jit/tensorexpr/reduction.h>
+
+#include <c10/util/irange.h>
+
+namespace torch {
+namespace jit {
+namespace tensorexpr {
+
+template <typename Op>
+static Expr* mutate_binary_op(
+    BinaryOpNode<Op>* v,
+    IRCloner* cloner,
+    bool option = false) {
+  Expr* lhs_new = v->lhs()->accept_mutator(cloner);
+  Expr* rhs_new = v->rhs()->accept_mutator(cloner);
+  IRNodeType expr_type = v->expr_type();
+  switch (expr_type) {
+    case IRNodeType::kAdd:
+      return new Add(lhs_new, rhs_new);
+    case IRNodeType::kSub:
+      return new Sub(lhs_new, rhs_new);
+    case IRNodeType::kMul:
+      return new Mul(lhs_new, rhs_new);
+    case IRNodeType::kDiv:
+      return new Div(lhs_new, rhs_new);
+    case IRNodeType::kMod:
+      return new Mod(lhs_new, rhs_new);
+    case IRNodeType::kMax:
+      return new Max(lhs_new, rhs_new, option);
+    case IRNodeType::kMin:
+      return new Min(lhs_new, rhs_new, option);
+    case IRNodeType::kAnd:
+      return new And(lhs_new, rhs_new);
+    case IRNodeType::kOr:
+      return new Or(lhs_new, rhs_new);
+    case IRNodeType::kXor:
+      return new Xor(lhs_new, rhs_new);
+    case IRNodeType::kLshift:
+      return new Lshift(lhs_new, rhs_new);
+    case IRNodeType::kRshift:
+      return new Rshift(lhs_new, rhs_new);
+    default:
+      throw unsupported_dtype();
```

Comment:
Nit: we probably should throw a different error here, it's not related to dtype.

Reply:
Yeah. Replaced with `unimplemented_lowering`.